controllers/nodehealthcheck_controller.go

```diff
@@ -495,13 +495,10 @@ func (r *NodeHealthCheckReconciler) deleteOrphanedRemediationCRs(nhc *remediatio
 
 	log.Info("Going to delete orphaned remediation CRs", "count", len(orphanedRemediationCRs))
 	for _, cr := range orphanedRemediationCRs {
-		nodeName := cr.GetName()
-		if cr.GetAnnotations() != nil && len(cr.GetAnnotations()[commonannotations.NodeNameAnnotation]) > 0 {
-			nodeName = cr.GetAnnotations()[commonannotations.NodeNameAnnotation]
-		}
+		nodeName := utils.GetNodeNameFromCR(cr)
```

Comment:
also fixes `[FAIL] Node Health Check CR Reconciliation with expected permanent node deletion [It] it should delete orphaned CR when node is deleted`